// gatsby-theme-document/src/components/RightSidebar.js
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React, { useState } from 'react';
import mediaqueries from '../styles/media';
import ListItem from './ListItem';
import { graphql, StaticQuery } from 'gatsby'

/* import BackgroundImage from 'gatsby-background-image';
//import styled from 'styled-components';
import Img from 'gatsby-image';
 */

const RightSidebar = ({  tableOfContents, location }) => {
  const [listItems] = useState(() => {
    if (!tableOfContents.items) {
      return [];
    }
    const mappedLinks = [];
    function mapLinks(items) {
      items.forEach(item => {
        mappedLinks.push(item);
        if (item.items) {
          mapLinks(item.items);
        }
      });
    }
    mapLinks(tableOfContents.items);
    return mappedLinks;
  });
  https://source.unsplash.com/1600x900/?beach
/*     const { desktop } = useStaticQuery(
      graphql`
        query {
          desktop: file(relativePath: { eq: "seamless-bg-desktop.jpg" }) {
            childImageSharp {
              fluid(quality: 90, maxWidth: 4160) {
                ...GatsbyImageSharpFluid_withWebp_tracedSVG
              }
            }
          }
        }
      `
    ) */
/*   render={data => {
  const imageData = data.desktop.childImageSharp.fluid */
  return (
    <RightSidebarWrapper>
      <RightSidebarNav>
        {/* <BackgroundImage
          Tag="section"
            className={className}
            fluid={imageData}
            backgroundColor={`#040e18`}
            >
              {children}
        </BackgroundImage> */}       

      <img src="https://source.unsplash.com/400x200/?tech" alt="" />
        <RightSidebarTitle>Contents of The Page</RightSidebarTitle>
        <RightSidebarList>
          {listItems.map(item => (
            <RightSidebarListItem key={location.pathname + item.url}>
              <ListItem location={location} item={item} />
            </RightSidebarListItem>
          ))}
        </RightSidebarList>
      </RightSidebarNav>
    </RightSidebarWrapper>
  );
  //}
 // }
};

const RightSidebarWrapper = styled.aside`
  display: none;
  flex: 0 0 16rem;
  font-size: 0.75rem;
  font-weight: 600;
  ${mediaqueries.desktop_medium_up`
    display: block
  `};
`;

const RightSidebarNav = styled.nav`
  position: fixed;
  top: 0;
  bottom: 0;
  overflow-x: hidden;
  overflow-y: auto;
  width: 16rem;
  height: 100%;
  padding: 7rem 1rem 0;
  background: ${p => p.theme.colors.sidebar};
  transition: 0.25s var(--ease-in-out-quad);
  transform: ${p => (p.navOpen ? `translateX(16rem)` : null)};
  ${mediaqueries.desktop_up`
    transform: translateX(0);
    padding: 6.6rem 0 1rem;
  `};
`;

const RightSidebarTitle = styled.p`
  margin-top: 0;
  font-size: 1.0rem;
  font-weight: 900;
  text-transform: uppercase;
`;

const RightSidebarList = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
  & ul {
    margin: 0;
    padding: 0;
    list-style: none;
  }
`;

const RightSidebarListItem = styled.li`
  margin: 0.3rem 0;
`;

RightSidebar.propTypes = {
  tableOfContents: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired
};

export default RightSidebar;
